app/models/aapb_record.rb

```diff
@@ -71,11 +71,12 @@ def organization_pb_core_name
   private
 
   def process_id(id)
-    raise "Unexpected GUID format" unless id =~ /^cpb-aacip(\/|_)\d{2,3}-\w{8,10}/
+    raise "Unexpected GUID format" unless id =~ /^cpb-aacip[\/_-]\d{2,3}-\w{8,10}/
+
     case id
     when /^cpb-aacip\/{1}/
       return id.sub(/\//, '_')
-    when /^cpb-aacip_{1}/
+    when /^cpb-aacip(_|-){1}/
```

Comment:
I think this whole case statement can be replaced with simply...
```
id.tr('/', '_')
```
... right? And then on top of that, aren't we moving toward using the dash rather than the underscore?... so it would actually be...
```
id.tr('/', '-')
```

Reply:
theoretically, but i have concerns because there will continue to be old transcripts on s3 with underscores which is where this id is eventually going to be used.

Reply:
for the command, what about

`id.gsub(/cpb-aacip[\/_]/, '-')` ?

as for `_`-filename-having transcripts on s3, if there are still such transcript files, I think we should migrate those to have dash filenames because I think we've invested in underscore/slash vs dash *not* being meaningful in other places in AAPB (by normalizing them to dashes on ingest, etc)

Reply:
The logic I see in `#process_id` goes like...
```
If the ID isn't the right format
  then raise an error
end if

if the ID has a slash
  then convert the slash to an underscore and return it
else if the ID has an underscore or a dash
  then return the id
else if the ID doesn't have a slash, underscore, or dash
  raise an error (this is redundant I think with the guard statement at the top)
end
```
I think this can all be preserved by keeping the guard statements at the top, and then just blindly converting any existing slashes to underscores, and not doing anything else otherwise. Am I missing something?

Reply:
@foglabs I think `id.gsub(/cpb-aacip[\/_]/, '-')` is literally the same thing as `id.tr('/_', '-')`... so it's a personal preference I guess. Unless I'm missing something.

Reply:
Here's an example...
```
# 3 ID variations
ids = ["cpb-aacip/1234", "cpb-aacip_1234", "cpb-aacip-1234"]

# Convert both / and _ to -
ids.map{ |id| id.tr('/_', '-') 

# Or convert only / to -
ids.map{ |id| id.tr('/', '-') }

# Or convert only / to _
ids.map{ |id| id.tr('/', '_') }
```

Reply:
And hell.. while we're trimming the fat... if you can get `#process_id` down to a 2-liner... you might as well just put the logic right into the constructor if that's the only place it's getting called 🤷

Reply:
@afred yeah `#tr` is good with me, I was just going off of the earlier reply that only covered / guids